Framework/API/inc/MantidAPI/IAlgorithm.h

```diff
@@ -28,6 +28,11 @@ namespace API {
  */
 using AlgorithmID = void *;
 
+/// The current state of the algorithm object
+enum class ExecutionState { Uninitialized, Initialized, Running, Finished };
+/// The validity of the results of the algorithm object
+enum class ResultState { NotFinished, Failed, Success };
```

Comment:
Yes, but one is the state of the algorithm, and the other is the state of the results from that algorithm.  I agree that if it is uninitialised it will always be not finished, but it can also be not finished if the algorithm has been initialised, or is running but not complete yet.  It comes down to the question someone is asking of the algorithm.  Sometimes you want the detaild state of the algorithm, and sometimes you just want to see what the state of the result iss.